packages/@aws-cdk/core/lib/resource.ts

```diff
@@ -14,6 +38,17 @@ export interface IResource extends IConstruct {
    * The stack in which this resource is defined.
    */
   readonly stack: Stack;
+
+  /**
+   * The environment this resource belongs to.
+   * For resources that are created and managed by the CDK
+   * (generally, those created by creating new class instances like Role, Bucket, etc.),
+   * this is always the same as the environment of the stack they belong to;
+   * however, for imported resources
+   * (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
+   * that might be different than the stack they were imported into.
+   */
+  readonly resourceEnv: ResourceEnvironment;
```

Comment:
`resourceEnv` is a mouthful (think `myResource.resourceEnv`) ? Did you consider just `env` or even just exposing `region` and `account` directly (`myResource.account` and `myResource.region`)?

Reply:
> Did you consider just exposing `region` and `account` directly (`myResource.account` and `myResource.region`)?

Yes, but Rico wanted it changed in https://github.com/aws/aws-cdk/pull/8280#discussion_r450668052

I can change it to `env` if that's what you prefer.

Reply:
I don't really understand @rix0rrr's motivation for another level of indirection, but I'm okay with it. Let's rename to `env`.

Reply:
Renamed to `env`.